src/UI/Component/Table/Factory.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/* Copyright (c) 2017 Nils Haagen <nhaageng@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Table;
+
+/**
+ * Table factory
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       The Presentation Table lists some data from the system in a pleasant way.
+	 *       The user should be able to get a quick overview over records in a
+	 *       dataset as simple as possible; for this, only most relevant fields of
+	 *       a record are being displayed at first glance. The deeper exploration of
+	 *       data should be as easy: Rows can be expanded to show more extensive fields
+	 *       and additional data that is less important for the identification of a record.
+	 *
+	 *       The aim of the Presentation Table is to represent a dataset as an entirety
+	 *       rather than a list of single rows. The table focusses on exploring this set,
+	 *       while the purpose of this exploration is known and supported.
+	 *       A single record does not necessarily reference a persistent entity like an ilObject,
+	 *       but can be derived and composed from all kind of sources.
+	 *
+	 *   composition: >
+	 *       The Presentation Table consists of a title, a slot for View Controls
+	 *       and Presentation Rows.
+	 *       Rows again will be prefixed by an expander-button and consist of a
+	 *       title, a subtitle and a choice of record-fields.
```

Comment:
Please outline specifically how columns are mapped to parts of this control, hidden column-handling, multi-actions (see full list in overleaf comment)

Reply:
The mapping is defined by the developer via the aformentioned "row mapping closure". The table does not attempt to handle hidden columns and multi-actions.